### PR TITLE
nixos/pipewire: fix wireplumber with system-wide

### PR DESCRIPTION
**Pull request by dnr**
###### Description of changes

I recently upgraded to 22.05 on my headless music server, which runs pipewire in system-wide mode. I understand this is not recommended by upstream, but it works fine for this purpose. The upgrade switched wireplumber for pipewire-media-server, and I had to make a few fixes to get wireplumber working. If there's a better way to do any of this, let me know.

The two main pieces are:

- Set `DBUS_SESSION_BUS_ADDRESS` = `unix:path=/run/dbus/system_bus_socket`: for some reason wireplumber insists on trying to start a user dbus without this.
- Add a lua config to disable alsa reservations and flatpak portal. The config files that come with wireplumber call out these two settings as ones that should be disabled if running system-wide, and indeed it didn't route audio to my alsa device without disabling reservations. (No idea what the flatpak portal thing does.)

(In my personal config, I added `after` and `wants` on `dbus.service` to be extra-sure, but I'm pretty sure those aren't needed since dbus uses socket activation.)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/desktops/pipewire/wireplumber.nix [hunk 1]
```diff
@@ -37,11 +37,19 @@ in
     environment.systemPackages = [ cfg.package ];
 
     environment.etc."wireplumber/main.lua.d/80-nixos.lua" = lib.mkIf (!pwUsedForAudio) {
-     text = ''
+      text = ''
```
Comment (K900) [inline nixos/modules/services/desktops/pipewire/wireplumber.nix:40]: Extra space snuck in here?

**Review by K900 — APPROVED** [review]
LGTM, socket activation should cover it here.

**Review by 4z3 — APPROVED** [review]
Confirming that this fixes system-wide pipewire on 22.05 (x86_64-linux), thanks :+1:

**Comment by Lassulus** [comment]
I guess we should backport this to 22.05 also?

File: nixos/modules/services/desktops/pipewire/wireplumber.nix (hunk 1 above)
Comment (4z3) [reply nixos/modules/services/desktops/pipewire/wireplumber.nix:40]: So sad this PR got merged with this extra space.

Comment (Lassulus) [reply nixos/modules/services/desktops/pipewire/wireplumber.nix:40]: seems intentional and correct

Comment (4z3) [reply nixos/modules/services/desktops/pipewire/wireplumber.nix:40]: Indeed. Apparently this just looked weird in my browser. I've just checked it locally.

Comment (dnr) [reply nixos/modules/services/desktops/pipewire/wireplumber.nix:40]: Indeed it was intentional. I copy/pasted this block and noticed it was off. Thanks for the quick review.

**Pull request merged**